api/build.gradle

```diff
@@ -126,6 +126,9 @@ pmdTest {
     ruleSetFiles = files("../config/pmd/test-rulesets.xml")
     ruleSets = [] // Remove built-in rule sets
 }
+spotbugsMain {
+    excludeFilter = file("../config/spotbugs/main-exclude.xml")
```

Comment:
What does this do?

Reply:
We haven't had to configure this for any other project - are you sure it's necessary?

Reply:
According to the spotbugs gradle plugin docs: "Current version of SpotBugs Gradle Plugin uses the same way with FindBugs Gradle Plugin to configure. Please check the document for FindBugsExtension." The findbugs extension docs provide these default values:
```
    findbugs {
        toolVersion = "2.0.1"
        sourceSets = [sourceSets.main]
        ignoreFailures = true
        reportsDir = file("$project.buildDir/findbugsReports")
        effort = "max"
        reportLevel = "high"
        visitors = ["FindSqlInjection", "SwitchFallthrough"]
        omitVisitors = ["FindNonShortCircuit"]
        includeFilter = file("$rootProject.projectDir/config/findbugs/includeFilter.xml")
        excludeFilter = file("$rootProject.projectDir/config/findbugs/excludeFilter.xml")
    }
```
So if it's not working without configuring the excludeFilter then we may need to rename the 'spotbugs' directory to be 'findbugs'.

Reply:
actually, I just realised that our filenames are different - using different files for main and test. Presumably that means none of our spotbugs exclusions were working until now?

Reply:
We had to add this in because the current spotBugsTest task only excludes for tests, whereas we need to ignore a spotbug check when running the actual build. 
Because I have added a log message 'log.debug("Retrieve claim by id {}", id);' in the read API, id is passed in as a path variable, spotbug returns an error on this as this can cause crlf injection in the logs (even though this is a UUID). The option were to:
- Exclude this code from spotbug checks
- Add some regex to the id to replace and crlf which make it bit messy
- Add pattern to logging so that it strips crlf
- Annotate with @Suppressfbwarnings